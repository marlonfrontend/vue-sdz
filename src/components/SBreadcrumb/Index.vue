<template>
  <div class="s-breadcrumb">
    <ul :class="['list', color]">
      <li v-for="({text, current, to, external}, index) in items" :key="index" class="item">
        <a v-if="!current" :href="to" :target="external ? '_blank' : '_self' " class="parent">
          {{ text }}
        </a>
        <span v-else>{{ text }}</span>
        <s-icon v-if="!isLast(index)" icon="sdz-chevron-right" size="14" />
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'SBreadcrumb',

  components: {
    SIcon: () => import('../SIcon/Index.vue').then(c => c.default)
  },

  props: {
    color: {
      type: String,
      validator: function (value) {
        return ['light', 'dark'].indexOf(value) !== -1
      }
    },

    items: { type: Array, required: true }
  },

  methods: {
    isLast(index){
      const position = this.items.length - 1;
      console.log(index, position);
      return position === index;
    }
  }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-breadcrumb {

  & > .list{
    list-style-type: none;
    display: flex;
  }

  & > .list > .item{
    display: inline-flex;
    font-size: .8rem;
    color: color(base, medium);

    & > .s-icon{
      margin: 2px 12px;
    }
  }


  & > .list > .item> .parent{
    //text-decoration: underline;
    color: color(base, medium);
  }

  & > .list > .item> .parent:hover{
    text-decoration: underline;
  }

  & > .list.dark > .item, & > .list.dark > .item > .parent{
    color: color(neutral, base);
  }
}
</style>
