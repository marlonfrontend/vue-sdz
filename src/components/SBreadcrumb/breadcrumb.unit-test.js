import { mount } from '@vue/test-utils'

import SBreadcrumb from './Index.vue'

/* eslint-disable */
describe('SBreadcrumb', () => {

  let cmp;

  beforeEach(() => {
    cmp = mount(SBreadcrumb, {
      propsData: {
        color: 'light',
        items: [
          {
            disabled: true,
            current: false,
            text: 'Parent Page',
            to: '#',
            external: false
          },
          {
            disabled: true,
            current: false,
            text: 'Sub-Parent Page 1',
            to: '#',
            external: false
          },
          {
            disabled: true,
            current: true,
            text: 'Sub-Parent Page 2',
            to: '#',
            external: false
          }
        ]
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check text li', () => {

    const content = cmp.find('div > ul > li')
    expect(content.exists()).toBe(true)
    expect(content.text()).toBe('Parent Page')
  })

  test('check icon', () => {

    const content = cmp.find('div > ul > li > i')
    expect(content.classes('sdz-chevron-right')).toBe(true)
    expect(content.classes()).toContain('sdz-chevron-right')
  })

  test('check not dark class', () => {
    
    const div = cmp.find('div > ul')
    expect(div.classes('dark')).toBe(false)
  })

})
