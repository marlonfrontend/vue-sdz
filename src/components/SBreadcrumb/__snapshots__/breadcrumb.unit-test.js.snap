// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SBreadcrumb has the expected html structure 1`] = `
<div
  class="s-breadcrumb"
>
  <ul
    class="list light"
  >
    <li
      class="item"
    >
      <a
        class="parent"
        href="#"
        target="_self"
      >
        
        Parent Page
      
      </a>
       
      <i
        class="s-icon icon sdz-chevron-right"
        style="font-size: 14px;"
      />
    </li>
    <li
      class="item"
    >
      <a
        class="parent"
        href="#"
        target="_self"
      >
        
        Sub-Parent Page 1
      
      </a>
       
      <i
        class="s-icon icon sdz-chevron-right"
        style="font-size: 14px;"
      />
    </li>
    <li
      class="item"
    >
      <span>
        Sub-Parent Page 2
      </span>
       
      <!---->
    </li>
  </ul>
</div>
`;
